src/Workspaces/Core/Desktop/Workspace/Host/TemporaryStorage/TemporaryStorageServiceFactory.CopiedMemoryMappedViewHandle.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+
+namespace Microsoft.CodeAnalysis.Host
+{
+    internal partial class TemporaryStorageServiceFactory
+    {
+        /// <summary>
+        /// This critical handle increments the reference count of a <see cref="SafeMemoryMappedViewHandle"/>. The view
+        /// will be released when it is disposed <em>and</em> all pointers acquired through
+        /// <see cref="SafeBuffer.AcquirePointer"/> (which this class uses) are released.
+        /// </summary>
+        /// <remarks>
+        /// <para><see cref="CriticalHandle"/> types are not reference counted, and are thus somewhat limited in their
+        /// usefulness. However, this handle class has tightly restricted accessibility and is only used by managed
+        /// code which does not rely on it counting references.</para>
+        /// </remarks>
+        private unsafe sealed class CopiedMemoryMappedViewHandle : CriticalHandleZeroOrMinusOneIsInvalid
+        {
```

Comment:
> but I can't think of any special cases or concerns that apply here.

Think about it this way.  This code will live for years or decades.  Different people with different levels of understanding will come across this code in the future and may end up not realizing these things and may end up breaking things in subtle ways.  

Defensively commenting our code to make it clear what is/isn't ok just helps ensure that people in the future don't mess things up.  Again, this is subtle stuff.  And these patterns and types you're using are *not* common the Roslyn codebase.  That means people don't have a natural built-in understanding of all these details.  For example, i was unaware that this code was finalizer-safe.  Making the documentation for this clear this up hurts no one and can only help the long term sustainability of this code.

Heck, it can be as simple as:  Clients of this code should take care to Dispose this type so that the memory mapped file can be released by teh OS as soon as it is no longer necessary.  However, if a client does not Dispose things, the GC will still take care of this by eventually finalizing this type.  All that means is that the memory mapped file will be kept alive longer than absolutely necessary.